File: pharmacy-api/public-api/controller/pickup_controller.go
package controller

import (
	"log"
	"net/http"

	"github.com/gin-gonic/gin"
	"github.com/go-redis/redis/v8"
)

func Pickup(redisDB *redis.Client) gin.HandlerFunc {
	return func(c *gin.Context) {

		key := c.Query("id")

		if key == "" {
			c.JSON(http.StatusBadRequest, gin.H{
				"Error": "key is required",
			})
			return
		}

		val, err := redisDB.HGetAll(c.Request.Context(), "public_task:"+key).Result()

		if len(val) == 0 {
			c.JSON(http.StatusNotFound, gin.H{
				"Status": "not_found",
			})
			return

		} else if err != nil {
			c.JSON(http.StatusInternalServerError, gin.H{
				"Error": "Internal error",
			})
			return

		}

		switch val["status"] {
		case "completed":
			err := redisDB.Del(c.Request.Context(), "public_task:"+key).Err()
			if err != nil {
				log.Printf("Error: %v", err)
				c.JSON(http.StatusOK, gin.H{
					"Status": val["status"],
					"Value":  val["result"],
				})
				return
			}
			c.JSON(http.StatusOK, gin.H{
				"Status": val["status"],
				"Value":  val["result"],
			})
			return
		case "pending":
			c.JSON(http.StatusOK, gin.H{
				"Status": val["status"],
			})
		case "error":
			c.JSON(http.StatusOK, gin.H{
				"Status": val["status"],
			})
		}

	}
}

File: pharmacy-api/src/local-api/controllers/pickup_controller.ts
import { Request, Response } from 'express';
import Redis from 'ioredis';

export function pickup(redisDB: Redis) {
  return async (req: Request, res: Response): Promise<void> => {
    const key = req.query.key as string;

    if (!key) {
      res.status(400).json({
        Error: "key is required",
      });
      return;
    }

    try {
      const val = await redisDB.hgetall(`local_task:${key}`);

      if (!val || Object.keys(val).length === 0) {
        res.status(404).json({
          Status: "not_found",
        });
        return;
      }

      switch (val.status) {
        case "completed":
          // Удаляем ключ после получения результата
          await redisDB.del(`local_task:${key}`).catch((err) => {
            console.error(`Error deleting key local_task:${key}:`, err);
          });
          
          res.status(200).json({
            Status: val.status,
            Value: val.result,
          });
          return;
          
        case "pending":
          res.status(200).json({
            Status: val.status,
          });
          return;
          
        case "error":
          res.status(200).json({
            Status: val.status,
          });
          return;
          
        default:
          res.status(200).json({
            Status: val.status,
          });
          return;
      }
    } catch (err) {
      console.error("Error in pickup controller:", err);
      res.status(500).json({
        Error: "Internal error",
      });
      return;
    }
  };
}